### DISCUSSION TITLE
Is there anyway I can block concurrent login for same user on other device?

### DISCUSSION THREAD

**Discussion by bhagyesh-7span** [directus/directus#17436] [Archive]
The options I have explored till now. 

- Explored using custom API hooks to delete user session from the directus_session table using user_id. This approach was met with a discussion [#15897](https://github.com/directus/directus/discussions/15897).

- Tried using socket.io to reload the page on the second login of the same user. Was able to make a connection from the server to an external HTML file but couldn't find a workaround for the frontend extension to attach the socket.io client.

- Currently looking for a way to achieve "**per user one session allowed**".

If anyone can provide input it would be helpful.

**Comment by paescuj** [comment]
You might be able to do something with the `auth.login` filter event: https://docs.directus.io/extensions/hooks.html#filter-events
For example, track the current session somewhere and block the login if there's already an active session.
However, I can imagine that you have to be careful with such an implementation so the active user isn't getting locked out as well.

**Reply by bhagyesh-7span** [reply]
Yes @paescuj I am able to do so and achieved using the same as you mentioned.
